src/extra/plugin/remote.js

```diff
@@ -264,9 +264,8 @@ window.ParsleyConfig.validators.remote = {
   priority: -1
 };
 
-$(document).on('form:submit.parsley', function (evt, parsleyForm) {
-  if (!evt.isDefaultPrevented())
-    parsleyForm._remoteCache = {};
+window.Parsley.on('form:submit', function () {
+  this._remoteCache = {};
```

Comment:
I think that if you remove cache but form submit is prevented because invalid, it could lack for future calls.
But was more a nice to have than an absolute requirement (dunno if it really occurs)